### PR TITLE
adds max_num_empty_steps parameter

### PR DESCRIPTION
**Pull request by bryano**
Controls how many steps to go without making any new logical gates possible. (Equivalent to #2619 when set to 1, and equivalent to prior code when set to infinity.)

### CODE REVIEW

**Comment by bryano** [comment]
@askolik I couldn't request your review for some reason.

File: cirq/contrib/routing/greedy.py [hunk 1]
```diff
@@ -284,22 +301,19 @@ def apply_next_swaps(self):
...
+        self.apply_next_swaps(True)
 
     def route(self):
         self.apply_possible_ops()
+        empty_steps_remaining = self.max_num_empty_steps
         while self.remaining_dag:
-            self.apply_next_swaps()
-            self.apply_possible_ops()
+            self.apply_next_swaps(not empty_steps_remaining)
+            if self.apply_possible_ops():
```
Comment (askolik) [inline cirq/contrib/routing/greedy.py]: Would pull out self.apply_possible_ops() of the if into a variable remaining_ops or similar for better readability.

Comment (bryano) [reply cirq/contrib/routing/greedy.py]: Done

File: cirq/contrib/routing/router_test.py [hunk 2]
```diff
@@ -97,6 +97,8 @@ def test_route_circuit_reproducible_between_runs(algo):
 (1, 2): ───0────X─────0────────────0─────iSwap────0───
                └──┘       └────┘       └──────┘
     """.lstrip('\n').rstrip()
+    print(expected_swap_network_str)
+    print(swap_network_str)
```
Comment (kevinsung) [inline cirq/contrib/routing/router_test.py]: Leftover print statements.

File: cirq/contrib/routing/greedy_test.py [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+    device_graph = nx.Graph([(x, y), (y, z)])
+    initial_mapping = {x: a, z: b}
+    route_circuit_greedily(circuit,
+                           device_graph,
+                           initial_mapping=initial_mapping,
+                           max_search_radius=1)
+
+
+def test_loop_break():
+    pass
```
Comment (kevinsung) [inline cirq/contrib/routing/greedy_test.py]: What's this for?

File: cirq/contrib/routing/greedy_test.py [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+def test_random_choice():
+    x, y, z = (cirq.NamedQubit(s) for s in 'xyz')
+    a, b = (cirq.NamedQubit(s) for s in 'ab')
+    circuit = cirq.Circuit(cirq.CZ(a, b))
+    device_graph = nx.Graph([(x, y), (y, z)])
+    initial_mapping = {x: a, z: b}
+    route_circuit_greedily(circuit,
+                           device_graph,
+                           initial_mapping=initial_mapping,
+                           max_search_radius=1)
```
Comment (kevinsung) [inline cirq/contrib/routing/greedy_test.py]: What is this testing? The name is not very descriptive.

Comment (bryano) [reply cirq/contrib/routing/greedy_test.py]: A different change that I decided to not include in this PR; removed.

File: cirq/contrib/routing/greedy_test.py (hunk 3 above)
Comment (bryano) [reply cirq/contrib/routing/greedy_test.py]: Vestige from when I structured things differently; removed.

File: cirq/contrib/routing/router_test.py (hunk 2 above)
Comment (bryano) [reply cirq/contrib/routing/router_test.py]: Removed.

**Pull request merged**